frontend/components/Domain/Recipe/RecipeDialogAddToShoppingList.vue

```diff
@@ -237,30 +237,40 @@ export default defineComponent({
           }
         });
 
+        let currentTitle = "";
+        const onHandIngs: ShoppingListIngredient[] = [];
         const shoppingListIngredientSections = shoppingListIngredients.reduce((sections, ing) => {
-          // if title append new section to the end of the array
           if (ing.ingredient.title) {
-            sections.push({
-              sectionName: ing.ingredient.title,
-              ingredients: [ing],
-            });
-            return sections;
+            currentTitle = ing.ingredient.title;
           }
 
-          // append new section if first
-          if (sections.length === 0) {
+          // If this is the first item in the section, create a new section
+          if (sections.length === 0 || currentTitle !== sections[sections.length - 1].sectionName) {
+            if (sections.length) {
+              // Add the on-hand ingredients to the last section
+              sections[sections.length - 1].ingredients.push(...onHandIngs);
+              onHandIngs.length = 0;
+            }
             sections.push({
-              sectionName: "",
-              ingredients: [ing],
+              sectionName: currentTitle,
+              ingredients: [],
             });
+          }
+
+          // Store the on-hand ingredients for later
+          if (ing.ingredient.food?.onHand) {
+            onHandIngs.push(ing);
             return sections;
           }
 
-          // otherwise add ingredient to last section in the array
+          // Add the ingredient to last section
```

Comment:
```suggestion
          // Add the ingredient to previous section
```

Using `previous` instead of `last` is more intuitive to me; `last` gives an implication of "the final one"